internal/pkg/scaffold/olm-catalog/package_manifest.go

```diff
@@ -0,0 +1,190 @@
+// Copyright 2019 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package catalog
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"sort"
+	"strings"
+
+	"github.com/operator-framework/operator-sdk/internal/pkg/scaffold"
+	"github.com/operator-framework/operator-sdk/internal/pkg/scaffold/input"
+
+	"github.com/ghodss/yaml"
+	olmregistry "github.com/operator-framework/operator-lifecycle-manager/pkg/controller/registry"
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/afero"
+)
+
+const PackageManifestFileExt = ".package.yaml"
+
+type PackageManifest struct {
+	input.Input
+
+	// CSVVersion is the version of the CSV being updated.
+	CSVVersion string
+	// Channel is CSVVersion's package manifest channel. If a new package
+	// manifest is generated, this channel will be the manifest default.
+	Channel string
+	// If ChannelIsDefault is true, Channel will be the package manifests'
+	// default channel.
+	ChannelIsDefault bool
+}
+
+var _ input.File = &PackageManifest{}
+
+// GetInput gets s' Input.
+func (s *PackageManifest) GetInput() (input.Input, error) {
+	if s.Path == "" {
+		lowerProjName := strings.ToLower(s.ProjectName)
+		// Path is what the operator-registry expects:
+		// {manifests -> olm-catalog}/{operator_name}/{operator_name}.package.yaml
+		s.Path = filepath.Join(OLMCatalogDir, lowerProjName,
+			lowerProjName+PackageManifestFileExt)
+	}
+	return s.Input, nil
+}
+
+var _ scaffold.CustomRenderer = &PackageManifest{}
+
+// SetFS is a no-op to implement CustomRenderer.
+func (s *PackageManifest) SetFS(_ afero.Fs) {}
+
+// CustomRender either reads an existing package manifest or creates a new
+// manifest and modifies it based on values set in s.
+func (s *PackageManifest) CustomRender() ([]byte, error) {
+	i, err := s.GetInput()
+	if err != nil {
+		return nil, err
+	}
+	path := filepath.Join(s.AbsProjectPath, i.Path)
+
+	pm := &olmregistry.PackageManifest{}
+	if _, err := os.Stat(path); err != nil && os.IsNotExist(err) {
+		pm = s.newPackageManifest()
+	} else if err == nil {
+		b, err := ioutil.ReadFile(path)
+		if err != nil {
+			return nil, errors.Wrapf(err, "read package manifest %s", path)
```

Comment:
Nit: For user facing error messages we should probably prefix them with a `failed to ...` like we do elsewhere:
```suggestion
			return nil, errors.Wrapf(err, "failed to read package manifest %s", path)
```